src/sys/socket/sockopt.rs

```diff
@@ -1001,6 +1001,133 @@ sockopt_impl!(
     bool
 );
 
+// Documented on `man 7 netlink`
+#[cfg(any(target_os = "android", target_os = "linux"))]
+sockopt_impl!(
+    /// Add socket to group
+    NetlinkAddMembership,
+    SetOnly,
+    libc::SOL_NETLINK,
+    libc::NETLINK_ADD_MEMBERSHIP,
+    libc::c_uint
+);
+#[cfg(any(target_os = "android", target_os = "linux"))]
+sockopt_impl!(
+    /// Drop socket from group
+    NetlinkDropMembership,
+    SetOnly,
+    libc::SOL_NETLINK,
+    libc::NETLINK_DROP_MEMBERSHIP,
+    libc::c_uint
+);
+#[cfg(any(target_os = "android", target_os = "linux"))]
+sockopt_impl!(
+    /// Enable packet info control messages
+    NetlinkPacketInfo,
+    Both,
+    libc::SOL_NETLINK,
+    libc::NETLINK_PKTINFO,
+    bool
+);
+#[cfg(any(target_os = "android", target_os = "linux"))]
+sockopt_impl!(
+    /// Send broadcast errors
+    NetlinkBroadcastError,
+    Both,
+    libc::SOL_NETLINK,
+    libc::NETLINK_BROADCAST_ERROR,
+    bool
+);
+#[cfg(any(target_os = "android", target_os = "linux"))]
+sockopt_impl!(
+    /// Avoid receiving 'ENOBUFS' errors
+    NetlinkNoEnobufs,
+    Both,
+    libc::SOL_NETLINK,
+    libc::NETLINK_NO_ENOBUFS,
+    bool
+);
+#[cfg(any(target_os = "android", target_os = "linux"))]
+sockopt_impl!(
+    /// Listen to notifications from all network namespaces
+    NetlinkListenAllNsid,
+    SetOnly,
+    libc::SOL_NETLINK,
+    libc::NETLINK_LISTEN_ALL_NSID,
+    bool
+);
+#[cfg(any(target_os = "android", target_os = "linux"))]
+sockopt_impl!(
+    /// Trim the extack if needed
+    NetlinkCapAck,
+    Both,
+    libc::SOL_NETLINK,
+    libc::NETLINK_CAP_ACK,
+    bool
+);
+#[cfg(any(target_os = "android", target_os = "linux"))]
+sockopt_impl!(
+    /// Always send extack
+    NetlinkExtAck,
+    Both,
+    libc::SOL_NETLINK,
+    libc::NETLINK_EXT_ACK,
+    bool
+);
+#[cfg(any(target_os = "android", target_os = "linux"))]
+sockopt_impl!(
+    /// Get strict check
+    NetlinkGetStrictCheck,
+    Both,
+    libc::SOL_NETLINK,
+    libc::NETLINK_GET_STRICT_CHK,
+    bool
+);
+
+/// List netlink groups
+/// The returned Vec contains a bit set in chunks of u32.
+#[cfg(any(target_os = "android", target_os = "linux"))]
+#[derive(Copy, Clone, Debug)]
+pub struct NetlinkListMemberships;
+
+#[cfg(any(target_os = "android", target_os = "linux"))]
+impl GetSockOpt for NetlinkListMemberships {
+    type Val = Vec<u32>;
```

Comment:
Defining this as `Vec` requires consumers to do allocations that they might not otherwise need to.  If you define it as `AsRef<[u8]>`, then that would eliminate an allocation.  see `AlgSetKey`.

Reply:
How would this work if an array doesn't fit the length reported by the kernel?
Should I return an error?

Reply:
I didn't understand this comment, so this is still not addressed in the new version